crates/core/tedge_actors/src/lib.rs

```diff
@@ -138,93 +174,175 @@
-//! # impl ServiceProvider<(),(),NoConfig> for SomeActorBuilder {
-//! #     fn add_peer(&mut self, peer: &mut impl ServiceConsumer<(), (), NoConfig>) {
-//! #         todo!()
-//! #     }
-//! # }
-//! #
-//! # impl ServiceConsumer<(),(),NoConfig> for SomeOtherActorBuilder {
-//! #     fn get_config(&self) -> NoConfig {
-//! #        todo!()
-//! #     }
-//! #     fn set_request_sender(&mut self, request_sender: DynSender<()>) {
-//! #         todo!()
-//! #     }
-//! #     fn get_response_sender(&self) -> DynSender<()> {
-//! #         todo!()
-//! #     }
-//! # }
+//! # #[derive(Debug)]
+//! # struct SomeInput;
+//! # #[derive(Debug)]
+//! # struct SomeOutput;
+//! # struct SomeConfig;
+//! /// An actor builder declares that it provides a service
+//! /// by implementing the `ServiceProvider` trait for the appropriate input, output and config types.
+//! impl ServiceProvider<SomeInput,SomeOutput,SomeConfig> for SomeActorBuilder {
+//!     /// Exchange two message senders with the new peer, so each can send messages to the other
+//!     ///
+//!     /// This actor has first to create a specific message sender for that peer,
+//!     /// possibly using the configuration `peer.get_config()` to adapt the service,
+//!     /// and then to give this new sender to the peer using `peer.set_config()`.
+//!     ///
+//!     /// Finally, this actor has to store the message sender returned by `peer.get_response_sender()`
+//!     /// to be able to send output messages.
+//!     fn add_peer(&mut self, peer: &mut impl ServiceConsumer<SomeInput,SomeOutput,SomeConfig>) {
+//!         todo!()
+//!     }
+//! }
 //!
-//! // An actor builder declares that it provides a service
-//! // by implementing the `ServiceProvider` trait for the appropriate input, output and config types.
-//! //
-//! // Here, `SomeActorBuilder: ServiceProvider<SomeInput, SomeOutput, SomeConfig>`
-//! let mut producer = SomeActorBuilder::default();
+//! /// An actor builder also declares that it is a consumer of other services required by it. This is done
+//! /// by implementing the `ServiceConsumer` trait for the appropriate input, output and config types.
+//! impl ServiceConsumer<SomeInput,SomeOutput,SomeConfig> for SomeOtherActorBuilder {
+//!     fn get_config(&self) -> SomeConfig {
+//!        todo!()
+//!     }
 //!
-//! // An actor builder also declares that it is a consumer of other services required by it. This is done
-//! // by implementing the `ServiceConsumer` trait for the appropriate input, output and config types.
-//! //
-//! // Here, `SomeOtherActorBuilder: ServiceConsumer<SomeInput, SomeOutput, SomeConfig>`
-//! let mut consumer = SomeOtherActorBuilder::default();
+//!     /// Update this actor with the sender where the service expects input messages to be sent
+//!     fn set_request_sender(&mut self, request_sender: DynSender<SomeInput>) {
+//!         todo!()
+//!     }
+//!
+//!     /// Tell the service where to send its output messages to this actor
```

Comment:
```suggestion
//!     /// Tell the service where to send its output messages of this actor
```

Reply:
I have the feeling that your are confused by the input/output terms. The *same* message is seen by the sender as an output message and by the receiver as an input message.

Would the following wording sound better to you?

```suggestion
//!     /// Tell the service where to send its messages so they will be received by this actor
```

Reply:
It's kind of clear, is not it kind of `registering the address` where to send the messages?